vassal-app/src/main/java/VASSAL/counters/GlobalCommand.java

```diff
@@ -299,7 +300,10 @@ else if ((target.targetCompare == GlobalCommandTarget.CompareMode.MATCH) ||
             command = command.append(((Deck)stack).maybeShuffle()); // If it's an always-shuffle deck, shuffle it.
             visitor.setSelectedCount(0);
           }
-          final List<GamePiece> pieces = stack.asList();
+          final List<GamePiece> pieces = new ArrayList<>(stack.asList());
+          if (stack instanceof Deck) {
+            Collections.reverse(pieces); // So that we deal from "top" of deck
+          }
```

Comment:
A slightly more efficient thing you could do here is copy the list only if you need to reverse it:
```
List<GamePiece> pieces = stack.asList();
if (stack instanceof Deck) {
  pieces = new ArrayList<>(pieces);
  Collections.reverse(pieces);
}
```

Reply:
done